### PR TITLE
[DATA-763] add retry option in workflow for buf

### PR DESCRIPTION
**Pull request by JohnN193**
Adds a [retry](https://github.com/marketplace/actions/retry-step) option on the `make buf` step in CI, to hopefully prevent tests from failing due to uncontrollable reasons

https://viam.atlassian.net/browse/DATA-749

### CODE REVIEW

File: .github/workflows/pullrequest.yml [hunk 1]
```diff
@@ -16,6 +16,6 @@ on:
 
 jobs:
   test:
-    uses: viamrobotics/slam/.github/workflows/test.yml@main
+    uses: viamrobotics/slam/.github/workflows/test.yml@DATA-763
```
Comment (JohnN193) [inline .github/workflows/pullrequest.yml:19]: will revert after testing

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ on:
 
 jobs:
   appimage:
-    uses: viamrobotics/slam/.github/workflows/appimage.yml@main
+    uses: viamrobotics/slam/.github/workflows/appimage.yml@DATA-763
```
Comment (JohnN193) [inline .github/workflows/main.yml:21]: will revert after testing

File: .github/workflows/appimage.yml [hunk 3]
```diff
@@ -56,9 +56,15 @@ jobs:
...
-      run: |
-        sudo -u testbot bash -lc 'cd slam/slam-libraries && make bufinstall buf'
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 5
+        max_attempts: 3
+        retry_on: timeout
+        shell: bash
+        command: |
+          chown -R testbot .
```
Comment (JohnN193) [inline .github/workflows/appimage.yml:66]: will this cause issues if this step has a retry?

**Comment by JohnN193** [comment]
Things seem to be running, just need to run tests manually to see what happens if a retry is triggered

**Comment by JohnN193** [comment]
Appears to be okay if a retry is triggered, based off [this test](https://github.com/viamrobotics/slam/actions/runs/3474838198/jobs/5808440512#step:4:149)

File: .github/workflows/appimage.yml (hunk 3 above)
Comment (Otterverse) [reply .github/workflows/appimage.yml:66]: No, that should be fine.

**Review by Otterverse — APPROVED** [review]
Looks good to me. I'll be curious to see if this works well for you, as I'll likely switch to something similar for RDK itself at some point.

**Comment by kkufieta** [comment]
> Appears to be okay if a retry is triggered, based off [this test](https://github.com/viamrobotics/slam/actions/runs/3474838198/jobs/5808440512#step:4:149)

[q] How did you test this? Could you trigger a retry somehow?

**Comment by JohnN193** [comment]
> 

I setup the workflow to run off this branch(the @DATA-763 bits), then reran the workflows manually until I saw the bug happen. luckily it was fairly quick to see a retry get triggered.

**Pull request merged**